api/src/main/java/net/kyori/adventure/util/ShadyPines.java

```diff
@@ -77,4 +85,44 @@ public static boolean equals(final double a, final double b) {
   public static boolean equals(final float a, final float b) {
     return Float.floatToIntBits(a) == Float.floatToIntBits(b);
   }
+
+  /**
+   * Create an immutable snapshot of the {@code original} list.
+   *
+   * @param original the original list
+   * @param <E> element type
+   * @return the immutable snapshot list
+   * @since 4.8.0
+   */
+  public static <E> @NonNull List<E> immutableList(final @NonNull Collection<E> original) {
+    requireNonNull(original, "original");
+    return Collections.unmodifiableList(new ArrayList<>(original));
+  }
+
+  /**
+   * Create an immutable snapshot list containing the elements of {@code original} array.
+   *
+   * @param original the original array
+   * @param <E> element type
+   * @return the immutable snapshot list
+   * @since 4.8.0
+   */
+  @SafeVarargs
+  public static <E> @NonNull List<E> immutableList(final @NonNull E@NonNull... original) {
+    requireNonNull(original, "original");
+    return Collections.unmodifiableList(Arrays.asList(original));
+  }
+
+  /**
+   * Create an immutable snapshot of the {@code original} map.
+   *
+   * @param original the original map
+   * @param <K> key type
+   * @param <V> value type
+   * @return the immutable snapshot map
+   * @since 4.8.0
+   */
+  public static <K, V> @NonNull Map<K, V> immutableMap(final @NonNull Map<K, V> original) {
+    return Collections.unmodifiableMap(new HashMap<>(original));
+  }
```

Comment:
Perhaps we should add a new class for all the collection-related methods?

Reply:
could do yeah -- that'd make sense with another PR you have open at the moment I think?